src/operations/search_indexes/drop.ts

```diff
@@ -0,0 +1,35 @@
+import type { Document } from 'bson';
+
+import type { Collection } from '../../collection';
+import type { Server } from '../../sdam/server';
+import type { ClientSession } from '../../sessions';
+import type { Callback } from '../../utils';
+import { AbstractOperation } from '../operation';
+
+export class DropSearchIndexOperation extends AbstractOperation<void> {
+  /** @internal */
+  constructor(private readonly collection: Collection<any>, private readonly name: string) {
+    super();
+  }
+
+  execute(server: Server, session: ClientSession | undefined, callback: Callback<void>): void {
+    const namespace = this.collection.fullNamespace;
+
+    const command: Document = {
+      dropSearchIndex: namespace.collection
+    };
+
+    if (typeof this.name === 'string') {
+      command.name = this.name;
+    }
+
+    server.command(namespace, command, { session }, err => {
```

Comment:
I thought about it, but if it's alright with you, we can hold off.  This will go away once we're using async await in favor of:
```typescript
await server.command(namespace, command, { session });
```

Reply:
We have added `connection.commandAsync` we can do the same for server, just a suggestion

Reply:
I thought about it - but it doesn't make much of a difference unless `operation.execute` is also async because then we'd only be then chaining, instead of callback handling.

However, we can't easily do the same for operation.execute because we already have it - but the asyncified form calls the`operation.execute` callback form, so it needs to be defined in the callback overload.

Reply:
This is why I asked about refactoring `operation.execute` to be async throughout the driver.

Reply:
Yea I agree, it is not much difference, you'd end up with `commandAsync().then` to invoke the callback. Using the xAsync alternatives makes it a bit easier for someone with no context to convert to `await` later, but all these operations are fairly thin so it's not a big deal